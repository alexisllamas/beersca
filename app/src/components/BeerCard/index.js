import React from 'react';

const BeerCard = beer => {
  return (
    <div className="card column beer is-half">
      <header className="card-header">
        <p className="card-header-title">{beer.name}</p>
        <span
          className={`tag ${
            beer.abv < 5
              ? 'is-success'
              : beer.abv < 10
                ? 'is-warning'
                : beer.abv < 20 ? 'is-danger' : 'is-black'
          }`}
        >
          {beer.abv}%
        </span>
      </header>
      <div className="card-content">
        <div className="content">{beer.description}</div>
        <div className="notification">{beer.brewers_tips}</div>
        <img className="beer-image" src={beer.image_url} alt="the beer" />
      </div>
    </div>
  );
};

export default BeerCard;
