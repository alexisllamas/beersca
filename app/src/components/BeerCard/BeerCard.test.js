import React from 'react';
import mount from 'enzyme/mount';
import BeerCard from '.';

describe('BeerCard component', () => {
  it('renders without crashing', () => {
    const beer = {
      name: 'Cerveza de Alexis',
      abv: 10,
      description: 'Cerveza bonita uwu',
      brewers_tips: 'tómala fría',
      image_url: 'img',
    };
    const wrapper = mount(<BeerCard {...beer} />);
    expect(wrapper).toMatchSnapshot();
  });
});
