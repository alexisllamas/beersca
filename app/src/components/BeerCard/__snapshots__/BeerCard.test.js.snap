// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard component renders without crashing 1`] = `
<BeerCard
  abv={10}
  brewers_tips="tómala fría"
  description="Cerveza bonita uwu"
  image_url="img"
  name="Cerveza de Alexis"
>
  <div
    className="card column beer is-half"
  >
    <header
      className="card-header"
    >
      <p
        className="card-header-title"
      >
        Cerveza de Alexis
      </p>
      <span
        className="tag is-danger"
      >
        10
        %
      </span>
    </header>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        Cerveza bonita uwu
      </div>
      <div
        className="notification"
      >
        tómala fría
      </div>
      <img
        alt="the beer"
        className="beer-image"
        src="img"
      />
    </div>
  </div>
</BeerCard>
`;
